app/src/components/report/LocationPicker.tsx

```diff
@@ -0,0 +1,222 @@
+    reportCoords,
+    updateReportCoords,
+    fillOutAddress,
+    initialMarkerState,
+    locationSubmitted,
+}: LocationPickerProps) {
+    /** Retrieve last user reported location for the view of the map upon loading page*/
+    function fetchUserLastLocation(): ViewStateType {
+        if (typeof window === 'undefined') return DEFAULT_COORDINATES;
+
+        if (isUndefined(reportCoords)) {
+            const lastReportedLocation = localStorage.getItem('lastReportedLocation');
+
+            if (lastReportedLocation) {
+                const { longitude, latitude } = JSON.parse(lastReportedLocation);
+                return { longitude, latitude, zoom: 10.5 };
+            } else {
+                return DEFAULT_COORDINATES;
+            }
+        } else {
+            return { ...reportCoords!, zoom: 10.5 };
+        }
+    }
+    // View Box of the map upon entering a page
+    const [viewState, setViewState] = useState<ViewStateType | {}>(fetchUserLastLocation);
+    // Disable the marker after user already submit location
+    const [draggableMarker, setDraggableMarker] = useState<boolean>(initialMarkerState);
+    // Ref to open modal alert
+    const modalRef = useRef<ModalHandle | null>(null);
+
+    function handleDragEnd(event: MarkerDragEvent) {
+        const { lng, lat } = event.lngLat;
+        updateReportCoords({ longitude: lng, latitude: lat });
+    }
+
+    // Track user location and put the marker there only if the location is not submitted
+    function handleGeolocate(event: GeolocateResultEvent) {
+        if (!draggableMarker) return;
+        const { longitude, latitude } = event.coords;
+        updateReportCoords({ longitude, latitude });
+    }
+
+    // Reset current location and get new location
+    function resetLocationAndAddress() {
+        setDraggableMarker(true);
+        fillOutAddress('');
+    }
+
+    // Translate coordinates to address
+    async function reverseGeolocation() {
+        if (isUndefined(reportCoords)) return;
+
+        const { longitude, latitude } = reportCoords!;
+
+        if (!longitude || !latitude) return;
+
+        const endpoint = 'mapbox.places';
+        try {
+            const response = await fetch(
+                `https://api.mapbox.com/geocoding/v5/${endpoint}/${longitude},${latitude}.json?types=address,poi,place&access_token=${MAPBOX_ACCESS_TOKEN}`
+            );
+            const json = await response.json();
+            const results = json.features[0];
+            fillOutAddress(results.place_name);
+            // Disable marker after submitted location
+            setDraggableMarker(false);
+        } catch (error) {
+            console.log(error);
+        }
+    }
+
+    return (
+        <div
+            style={{
+                height: '100%',
+                width: '100%',
+            }}
+        >
+            <Map
+                mapboxAccessToken={MAPBOX_ACCESS_TOKEN}
+                reuseMaps={true} //remove this if map breaks
```

Comment:
```
//remove this if map breaks
```

Could you just share what this means? Sounds scary :)

Reply:
According to this [doc](https://visgl.github.io/react-map-gl/docs/api-reference/map#:~:text=reuseMaps%3A%20boolean%E2%80%8B,triggered%20by%20initialization.https://visgl.github.io/react-map-gl/docs/api-reference/map#:~:text=reuseMaps%3A%20boolean%E2%80%8B,triggered%20by%20initialization.), it looks like you're using it to reduce number of times Mapbox's map is being rendered so we can save money.
In the create/edit page, the user can switch between manual and map mode, causing the map to re-render, so I thought it would be good to have.
The note was left for myself in case the map breaks. It hasn't happened so I'll remove this comment :D